app/pages/creator-landing-page/[pubKey].tsx

```diff
@@ -51,7 +72,12 @@ const CreatorLandingPage = () => {
     );
     console.log(creatorPubKey.toBase58());
 
-    const creatorAccount = await program!.account.creator.fetch(creatorPubKey);
+    // route to different page if creator is not found from pubkey, create a new component from it
+    const creatorAccount = await ( pubKey == creatorPubKey.toBase58() ? program!.account.creator.fetch(pubKey) : null);
```

Comment:
I just realized that this logic is wrong. This component should take the `pubKey` from the url and fetch that Creator account. The connected wallet shouldn't matter in this case